p2p/client.go

```diff
@@ -13,15 +13,32 @@ import (
 	"github.com/libp2p/go-libp2p-core/peer"
 	discovery "github.com/libp2p/go-libp2p-discovery"
 	dht "github.com/libp2p/go-libp2p-kad-dht"
-	"github.com/libp2p/go-libp2p/p2p/host/routed"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	routedhost "github.com/libp2p/go-libp2p/p2p/host/routed"
 	"github.com/multiformats/go-multiaddr"
+	"go.uber.org/multierr"
 
 	"github.com/lazyledger/optimint/config"
 	"github.com/lazyledger/optimint/log"
 )
 
-// reAdvertisePeriod defines a period after which P2P client re-attempt advertising namespace in DHT.
-const reAdvertisePeriod = 1 * time.Hour
+// TODO(tzdybal): refactor to configuration parameters
+const (
+	// reAdvertisePeriod defines a period after which P2P client re-attempt advertising namespace in DHT.
+	reAdvertisePeriod = 1 * time.Hour
+
+	// peerLimit defines limit of number of peers returned during active peer discovery.
+	peerLimit = 60
+
+	// txTopicSuffix is added after namespace to create pubsub topic for TX gossiping.
+	txTopicSuffix = "-tx"
+)
+
+// TODO(tzdybal): refactor. This is only a stub.
+type Tx struct {
+	Data []byte
+}
```

Comment:
IMO, we will most likely keep it that way or even make it an alias like in tendermint: https://github.com/tendermint/tendermint/blob/9ecfcc93a6364bbecc9e1d7740b53602eda667eb/types/tx.go#L14-L17

Reason is the abstraction between tendermint (and with it optimint) and the abci app (that understands the Tx). 

It is certainly not the right place for the type definition though but that will be part of that refactoring mentioned in the todo I guess.